Cybersecurity_Tools/ClipCrypt/clipcrypt/encryption.py

```
@@ -0,0 +1,165 @@
+            ValueError: If decryption fails
+        """
+        try:
+            aesgcm = self._get_aesgcm()
+            ciphertext = base64.b64decode(encrypted_data)
+            nonce_bytes = base64.b64decode(nonce)
+            
+            # Decrypt the data
+            plaintext = aesgcm.decrypt(nonce_bytes, ciphertext, None)
+            return plaintext.decode('utf-8')
+        except Exception as e:
+            raise ValueError(f"Decryption failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)